File: AdventOfCode/2025/Day5/Python version/Part2.py
# Part 1
with open("input.txt", "r") as file:
    lines = [line.strip() for line in file]

try:
    empty = lines.index("")
except ValueError:
    empty = len(lines)

fresh = []
for i in range(empty):
    try:
        start, end = map(int, lines[i].split("-"))
        fresh.append((start, end))
    except ValueError:
        continue

if not fresh:
    print(0)
else:
    fresh.sort()
    merged = [fresh[0]]

    for start, end in fresh[1:]:
        last_start, last_end = merged[-1]
        if start <= last_end + 1:
            merged[-1] = (last_start, max(last_end, end))
        else:
            merged.append((start, end))

    total = sum(end - start + 1 for start, end in merged)
    print(total)

File: AdventOfCode/2025/Day5/Part2.cpp
// Part 2
#include <bits/stdc++.h>
#include <fstream>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    ifstream file("input.txt");
    if (!file.is_open()) return cerr << "ERROR", 1;

    vector<pair<long long, long long>> ranges;
    string line;

    while (getline(file, line)) {
        if (line.empty()) break;
        size_t dash = line.find('-');
        long long start = stoll(line.substr(0, dash));
        long long end = stoll(line.substr(dash + 1));
        ranges.emplace_back(start, end);
    }

    sort(ranges.begin(), ranges.end());

    vector<pair<long long, long long>> merged;
    for (auto& r : ranges) {
        if (merged.empty() || merged.back().second < r.first) {
            merged.push_back(r);
        } else {
            merged.back().second = max(merged.back().second, r.second);
        }
    }

    long long counter = 0;
    for (auto& r : merged) {
        counter += (r.second - r.first + 1);
    }

    cout << counter << '\n';
    return 0;
}